crosscluster/logical: TestRandomTables failed

crosscluster/logical.TestRandomTables [failed](https://mesolite.cluster.engflow.com/invocations/default/ae38f9c4-d792-4758-af10-566a371344c7?testReportRun=1&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL2xvZ2ljYWw6bG9naWNhbF90ZXN0) on master @ [f9bcbc4fdf0ba5236f6312c00becf303c29f98f9](https://github.com/cockroachdb/cockroach/commits/f9bcbc4fdf0ba5236f6312c00becf303c29f98f9):


```
=== RUN   TestRandomTables
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestRandomTables1373538763
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    logical_replication_job_test.go:1027: CREATE TABLE rand_table_0 (col1_0 DATE NOT NULL, col1_1 NAME NOT NULL, col1_2 BIT(39) NULL, col1_3 INT4 NOT NULL, col1_4 REFCURSOR, col1_5 TIMETZ NOT NULL, col1_6 VECTOR NOT NULL, col1_7 STRING NOT NULL AS (lower(CAST(col1_2 AS STRING))) STORED, col1_8 STRING NOT NULL AS (lower(CAST(col1_6 AS STRING))) VIRTUAL, col1_9 STRING NOT NULL AS (lower(CAST(col1_6 AS STRING))) STORED, col1_10 STRING NOT NULL AS (lower(CAST(col1_6 AS STRING))) VIRTUAL, col1_11 STRING AS (lower(CAST(col1_4 AS STRING))) VIRTUAL, col1_12 INT4 NOT NULL AS (abs(col1_3)) VIRTUAL, PRIMARY KEY (col1_3 ASC, col1_7 ASC), UNIQUE (col1_5 ASC), UNIQUE (col1_8), INDEX (lower(CAST(col1_5 AS STRING)) DESC, col1_0 DESC, lower(CAST(col1_4 AS STRING)) ASC) STORING (col1_1, col1_4, col1_5), INDEX (col1_7 ASC, col1_2 ASC))
    logical_replication_job_test.go:1034: pq: cannot create logical replication stream: RefCursor is not supported by LDR
    logical_replication_job_test.go:1027: CREATE TABLE rand_table_1 (col1_0 JSONB NOT NULL, col1_1 STRING COLLATE en_US NOT NULL, col1_2 OID NULL, PRIMARY KEY (col1_1 ASC), INVERTED INDEX (col1_1 DESC, col1_0), INDEX (col1_1 ASC, col1_0 DESC, col1_2 DESC))
    logical_replication_job_test.go:1034: pq: cannot create logical replication stream: table rand_table_1 has a primary key column (col1_1) with composite encoding
    logical_replication_job_test.go:1027: CREATE TABLE rand_table_2 (col1_0 "char" NOT NULL, col1_1 BYTES NOT NULL, col1_2 JSONB NOT NULL, col1_3 TIMETZ NULL, col1_4 STRING NOT NULL AS (lower(col1_0)) STORED, col1_5 STRING NOT NULL AS (lower(col1_0)) VIRTUAL, col1_6 STRING NULL AS (lower(CAST(col1_3 AS STRING))) STORED, PRIMARY KEY (col1_5 DESC), INDEX (col1_4 ASC, col1_3 ASC, col1_6 ASC), UNIQUE (lower(col1_0) ASC) STORING (col1_2, col1_3, col1_6), INDEX (col1_2, col1_6 DESC, col1_0 DESC, col1_5 DESC), UNIQUE (col1_4), UNIQUE (col1_2, col1_4), UNIQUE (col1_5 DESC) STORING (col1_0, col1_3), INDEX (col1_4) STORING (col1_3, col1_6))
    logical_replication_job_test.go:1034: pq: cannot create logical replication stream: table rand_table_2 has a virtual computed column col1_5 that appears in the primary key
    logical_replication_job_test.go:1027: CREATE TABLE rand_table_3 (col1_0 STRING NOT NULL, col1_1 FLOAT8 NOT NULL, col1_2 OID NOT NULL, col1_3 BPCHAR NOT NULL, col1_4 BIT(2) NOT NULL, col1_5 LTREE NOT NULL, col1_6 GEOMETRY, col1_7 STRING AS (lower(CAST(col1_6 AS STRING))) STORED, col1_8 STRING NULL AS (lower(col1_0)) VIRTUAL, col1_9 STRING AS (lower(CAST(col1_6 AS STRING))) VIRTUAL, col1_10 FLOAT8 NOT NULL AS (abs(col1_1)) STORED, PRIMARY KEY (col1_0 DESC, col1_2 ASC), INDEX (col1_0 ASC), INVERTED INDEX (col1_1 ASC, col1_7, col1_9 gin_trgm_ops ASC), INDEX (col1_2 ASC) NOT VISIBLE, INVERTED INDEX (col1_9 gin_trgm_ops ASC), INDEX (col1_0))
    utils.go:53: waiting for logical replication job 1165998151789871105 to reach replicated time of 1775904797.911584865,0
    utils.go:55: condition failed to evaluate within 3m45s: from utils.go:70: logical replication has not recorded any progress yet, waiting to advance past 1775904797.911584865,0
        
        goroutine dump: /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/testlogs/pkg/crosscluster/logical/logical_test/shard_3_of_5_run_1_of_3/test.outputs/goroutine_dump_1817425723.txt
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestRandomTables1373538763
--- FAIL: TestRandomTables (757.44s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62845

--- a/pkg/crosscluster/logical/logical_replication_job_test.go
+++ b/pkg/crosscluster/logical/logical_replication_job_test.go
@@ -997,6 +997,7 @@ func TestFilterRangefeedInReplicationStream(t *testing.T) {
 func TestRandomTables(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	skip.UnderDeadlock(t)
+	skip.UnderRace(t)
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()

